python-scripts/dump_ram.py

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env python3
+
+# Read all physical memory
+
+from xboxpy import *
+
+ram_size = 64 * 1024 * 1024
+
+f = open('ram.bin', 'wb')
+
+offset = 0
+chunk_size = 0x1000
+assert(ram_size % chunk_size == 0)
+while offset < ram_size:
+  mapped = ke.MmMapIoSpace(offset, chunk_size, ke.PAGE_READWRITE)
```

Comment:
What if the offset was already mapped? Would that create another mapping here, or would the call fail?

Reply:
It maps physical pages to non-paged system space (should be 0xD... on Xbox).
So you can map as often as you like, as long as there's enough non-paged virtual space available.
You can also use this to map pages with different permissions / caching.

Also see https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/content/wdm/nf-wdm-mmmapiospace
*(Which has another 3rd argument? I should probably be looking into this)*